packages/protocol/contracts/rrp/authorizers/RequesterRrpAuthorizer.sol

```diff
@@ -5,28 +5,66 @@ import "../../adminnable/Whitelister.sol";
-    /// contracts may require to access the arguments that are redundant here
-    /// @param requestId Request ID
+    /// @notice Called to check if a user is whitelisted to use the
+    /// Airnode–endpoint pair
     /// @param airnode Airnode address
     /// @param endpointId Endpoint ID
-    /// @param sponsor Sponsor address
-    /// @param requester Requester address
-    /// @return Authorization status of the request
-    function isAuthorized(
-        bytes32 requestId, // solhint-disable-line no-unused-vars
+    /// @param user User address
+    /// @return isWhitelisted If the user is whitelisted
+    function userIsWhitelisted(
         address airnode,
-        bytes32 endpointId, // solhint-disable-line no-unused-vars
-        address sponsor, // solhint-disable-line no-unused-vars
-        address requester
-    ) external view override returns (bool) {
-        return userIsWhitelisted(bytes32(abi.encode(airnode)), requester);
+        bytes32 endpointId,
+        address user
+    ) external view override returns (bool isWhitelisted) {
+        return userIsWhitelisted(deriveServiceId(airnode, endpointId), user);
+    }
+
+    /// @notice Called to get the detailed whitelist status of a user for the
+    /// Airnode–endpoint pair
+    /// @param airnode Airnode address
+    /// @param endpointId Endpoint ID
+    /// @param user User address
+    /// @return expirationTimestamp Timestamp at which the whitelisting of the
+    /// user will expire
+    /// @return whitelistedPastExpiration Whitelist status that the user will
+    /// have past expiration
+    function airnodeToEndpointIdToUserToWhitelistStatus(
+        address airnode,
+        bytes32 endpointId,
+        address user
+    )
+        external
+        view
+        override
+        returns (uint64 expirationTimestamp, bool whitelistedPastExpiration)
+    {
+        WhitelistStatus
+            storage whitelistStatus = serviceIdToUserToWhitelistStatus[
+                deriveServiceId(airnode, endpointId)
+            ][user];
+        expirationTimestamp = whitelistStatus.expirationTimestamp;
+        whitelistedPastExpiration = whitelistStatus.whitelistedPastExpiration;
+    }
+
+    /// @notice Called internally to derive the service ID of the
+    /// Airnode–endpoint pair
+    /// @dev Whitelister contract that this contract inherits keeps whitelist
+    /// statuses in a single level hash map. We have two parameters here
+    /// (Airnode address and endpoint ID) from which we need to derive a single
+    /// service ID, and we do this by calculating their hash.
+    /// @param airnode Airnode address
+    /// @param endpointId Endpoint ID
+    /// @return serviceId Service ID
+    function deriveServiceId(address airnode, bytes32 endpointId)
+        internal
+        pure
+        returns (bytes32 serviceId)
+    {
+        return keccak256(abi.encodePacked(airnode, endpointId));
```

Comment:
this is more of a question but if we use encode we can unpack the contents later. Wouldn't that be useful instead of just hashing it?

Reply:
We can't unpack the contents of the encoding because it's being hashed. It doesn't matter if `abi.encode` or `abi.encodePacked` is used here, and in these cases I use `encodePacked`, see https://github.com/api3dao/airnode/blob/f09794864e3daf58a4bb41dbc6e35d1d1c0c6dcf/packages/protocol/contracts/rrp/AirnodeRrp.sol#L130 (I think this changed with beta)

Reply:
My question is why are we hashing it, is it not possible to just encode and return the result?

Reply:
Because we need an identifier of `bytes32` type. `abi.encode` returns a `bytes` type that is longer than 32 bytes.

Reply:
I see thanks.